### PR TITLE
Refactoring/gcp/cloudresourcemanager

### PR DESCRIPTION
**Pull request by Remi05**
Migrated GCP's Cloud Resource Manager service to the new architecture.

Note that `gcp/resources/projects.py` and `gcp/facade/base.py` are the same as in the [Stackdriver logging PR](https://github.com/nccgroup/ScoutSuite/pull/229) and the [IAM PR](https://github.com/nccgroup/ScoutSuite/pull/230); they are simply shared by all services and will appear in all GCP refactoring PRs until one of the services is merged.

### CODE REVIEW

**Comment by j4v** [comment]
:clap: :clap: :clap:

File: ScoutSuite/providers/gcp/resources/cloudresourcemanager/bindings.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            return members_dict
+
+        type_map = { 
+            'user': 'users', 
+            'group': 'groups', 
+            'serviceAccount': 'service_accounts'
+        }
+        
+        for member in raw_binding['members']:
+            member_type, entity = member.split(':')[:2]
```
Comment (Aboisier) [inline ScoutSuite/providers/gcp/resources/cloudresourcemanager/bindings.py]: Maybe add a comment explaining this line, it's a bit esoteric if you don't know what the raw object looks like.

File: ScoutSuite/providers/gcp/facade/cloudresourcemanager.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+
```
Comment (misg) [inline ScoutSuite/providers/gcp/facade/cloudresourcemanager.py]:
```suggestion
from ScoutSuite.providers.utils import run_concurrently
```

File: ScoutSuite/providers/gcp/facade/cloudresourcemanager.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from googleapiclient import discovery
+from ScoutSuite.providers.gcp.utils import MemoryCache
+
+class CloudResourceManagerFacade:
+    def __init__(self):
+        self._resourcemanager_client = discovery.build('cloudresourcemanager', 'v1', cache_discovery=False, cache=MemoryCache())
+
+    # TODO: Make truly async
+    async def get_bindings(self, project_id):
+        return self._resourcemanager_client.projects().getIamPolicy(resource=project_id).execute()
```
Comment (misg) [inline ScoutSuite/providers/gcp/facade/cloudresourcemanager.py]:
```suggestion
        return await run_concurrently(
                lambda: self._resourcemanager_client.projects().getIamPolicy(resource=project_id).execute()
        )
```

**Comment by Remi05** [comment]
Added left menu for projects and fixed UI.

Before : 

cloudresourcemanager_before

After : 

cloudresourcemanager_after

File: ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+class CloudResourceManager(Projects):
+    _children = [  
+        (Bindings, 'bindings') 
+    ]
+
+    def __init__(self, gcp_facade: GCPFacade):
+        super(CloudResourceManager, self).__init__(gcp_facade)
+
+
```
Comment (Aboisier) [inline ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py]:
```suggestion
```

File: ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+from ScoutSuite.providers.gcp.resources.projects import Projects
+from ScoutSuite.providers.gcp.resources.cloudresourcemanager.bindings import Bindings
+
+class CloudResourceManager(Projects):
+    _children = [  
+        (Bindings, 'bindings') 
+    ]
+
+    def __init__(self, gcp_facade: GCPFacade):
+        super(CloudResourceManager, self).__init__(gcp_facade)
```
Comment (Aboisier) [inline ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py]: Is this constructor necessary? It's the same as the base.

Comment (Remi05) [reply ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py]: Yeah, after some additional refactoring it became obsolete. Good catch!

**Pull request merged**